concepts/pointers/introduction.md

```diff
@@ -1,3 +1,149 @@
-# Introduction
+# About
+
+Like many other languages, Go has pointers. A pointer hold the memory address of a value.
+
+## Variables and Memory
+
+Let's say we have a regular integer variable `a`:
+
+```go
+var a int
+```
+
+When we declare a variable, Go has to find a place in memory to place its value. Conviniently, when we need to fetch the value stored in that piece of memory, we can just refer to it by the variable name.
+
+For instance, when we do:
+
+```go
+a + 2
+```
+
+We are effectively fetching the value stored in the memory associated with the variable `a` and adding 2 to it.
+
+Similarly, when we need to change the value in the piece of memory of `a`, we can use the variable name to do an assignment:
+
+```go
+a = 3
+```
+
+The piece of memory that `a` is associated with now will have the value `3`.
+
+## Pointers
+
+While variables allow us to refer to values in memory, sometimes it's useful to know the memory address of the value that the variable refers. Pointers hold the memory addresses of those values. A pointer declaration looks like this:
+
+```go
+var p *int // 'p' can hold the memory address of an integer
+```
+
+Here we declaring a variable `p` of type `*int`. The type `*int` means "pointer to int". In other words, we are saying that `p` is a pointer to int, which means that `p` will hold the memory address of an integer. The zero value of pointers is `nil`. A `nil` pointer holds no memory address.
```

Comment:
```suggestion
Here we declaring a variable `p` of type `*int`. The type `*int` means "pointer to int". In other words, we are saying that `p` is a pointer to int, which means that `p` will hold the memory address of an integer. The zero value of pointers is `nil` because a `nil` pointer holds no memory address.
```

Reply:
I felt those two sentences should be linked so it feels a little less like two seperate facts.